aten/src/ATen/native/cpu/Vec256.h

```diff
@@ -41,29 +41,34 @@ namespace vec256 {
 template <class T> class Vec256 {
 public:
   T values[32 / sizeof(T)]; // Mimics AVX behavior
-  inline void load(const T *ptr) { 
-    std::memcpy(values, ptr, 32); 
-  };
-  inline void store(T *ptr) { std::memcpy(ptr, values, 32); }
-  inline size_t size() { return 32 / sizeof(T); }
+  inline void load(const T *ptr) { std::memcpy(values, ptr, 32); };
+  inline void store(T *ptr) const { std::memcpy(ptr, values, 32); }
+  static constexpr size_t size = 32 / sizeof(T);
   inline void operator=(const Vec256<T> &b) {
     std::memcpy(values, b.values, 32);
   }
+  inline void map(T (*f)(T)) {
+    for (size_t i = 0; i < size; i++)
+      values[i] = f(values[i]);
+  }
+  inline void ceil() { map(std::ceil); }
```

Comment:
Make these functions return a new Vec instead of modifying it in-place.

Reply:
Why?

Reply:
For the same reason that `std::ceil(float)` returns a float instead of modifying a reference: It's easier to reason about math functions that return new values.